objects/src/accounts/delta/storage.rs

```diff
@@ -20,224 +26,135 @@ const MAX_MUTABLE_STORAGE_SLOT_IDX: u8 = 254;
 /// - item updates: represented by `cleared_items` and `updated_items` field.
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 pub struct AccountStorageDelta {
-    pub cleared_items: Vec<u8>,
-    pub updated_items: Vec<(u8, Word)>,
-    pub updated_maps: Vec<(u8, StorageMapDelta)>,
+    slots: BTreeMap<u8, Word>,
+    maps: BTreeMap<u8, StorageMapDelta>,
 }
 
 impl AccountStorageDelta {
+    /// Creates a new storage delta from the provided fields.
+    pub const fn new(slots: BTreeMap<u8, Word>, maps: BTreeMap<u8, StorageMapDelta>) -> Self {
+        Self { slots, maps }
+    }
```

Comment:
Can we combine `AccountStorageDelta` and `AccountStorageDeltaTracker`? They seem like the exact same thing.